docs/node/migrations/zero-downtime-migrations/migration-proposal.mdx

```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 1
+sidebar_label: "Migration Proposal"
+id: migration-proposal
+title: Migration Proposal
+description: "Migration Proposal"
+slug: /node/migrations/zero-downtime-migrations/migration-proposal
+---
+
+Migration Declaration is a network governance proposal to coordinate the network migration process. The migration proposal includes the activation period, duration, and the new chain ID that the network will migrate to.
+
+```go
+// MigrationDeclaration creates a new migration. It is used to agree on terms of a migration,
+// and is voted on using Kwil's vote store.
+type MigrationDeclaration struct {
+	// ActivationPeriod is the amount of blocks before the migration is activated.
+	// It starts after the migration is approved via the voting system.
+	// The intention is to allow validators to prepare for the migration.
+	ActivationPeriod uint64
+	// Duration is the amount of blocks the migration will take to complete.
+	Duration uint64
+	// ChainID is the new chain ID that the network will migrate to.
+	// A new chain ID should always be used for a new network, to avoid
+	// cross-network replay attacks.
+	ChainID string
+}
+```
+
+The Migration proposal needs to be approved by atleast 2/3rd of the validators to start the migration process. The migration proposal will get expired if the supermajority of the validators do not approve the proposal within `100800` blocks (roughly 7 days considering the block rate of 6sec). Once the migration proposal is approved, the network moves into migration mode after the specified activation period.
+
+```
+// MigrationResolution is the definition for the network migration vote type in Kwil's
+// voting system.
+var MigrationResolution = resolutions.ResolutionConfig{
+	ConfirmationThreshold: big.NewRat(2, 3),
+	ExpirationPeriod:      100800, // 1 week
+	ResolveFunc: func(ctx context.Context, app *common.App, resolution *resolutions.Resolution, block *common.BlockContext) error {
+		return startMigration(ctx, app, resolution, block)
+	},
+}
+```
+
+## Submit Migration Proposal
+
+A validator operator can submit a migration proposal to upgrade the network using the `kwil-admin migrate propose` command as shown below. Refer to the [kwil-admin migrate](/docs/ref/kwil-admin/migration/propose) documentation for more details.
+
+```bash
+kwil-admin migrate propose --activation-period <activationPeriod> --duration <migrationDuration> --chain-id <newChainID> --rpcserver /tmp/kwil.sock
+```
+
+:::note
+An off-chain communication is required among the validator operators to decide on the details of the migration proposal. One of the validators should submit the proposal to the network for approval. Only one proposal can be applied to the network during the lifetime of the network. If multiple validators submit the same proposal, the network will only consider the first proposal submitted.
```

Comment:
Is this true even if the network rejects the proposal? For example, a proposal is submitted and then expires. Can we submit a new proposal?